app/controllers/products_controller.rb

```diff
@@ -0,0 +1,61 @@
+class ProductsController < ApplicationController
+  before_action :admin_authorization, only: %i(new create edit destroy update)
+  before_action :load_product, only: %i(show edit destroy update)
+
+  def new
+    @product = Product.new
+  end
+
+  def show
+    @is_admin = is_admin?
+  end
+
+  def create
+    @product = Product.new(product_params)
+
+    if @product.save
+      redirect_to @product
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @product.update(product_params)
+      redirect_to @product
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @product.destroy
+    redirect_to root_path, status: :see_other
```

Comment:
What does status `:see_other` mean?

Reply:
The 303 See Other code is typically provided in response to a POST, PUT, or DELETE.
indicates that the redirects does not link to the requested resource itself, but to another page